README.md

```diff
@@ -427,17 +418,17 @@ stacks:
 stack_master help # Display up to date docs on the commands available
 stack_master init # Initialises a directory structure and stack_master.yml file
 stack_master list # Lists stack definitions
-stack_master apply [region-or-alias] [stack-name] # Create or update a stack
-stack_master apply [region-or-alias] [stack-name] [region-or-alias] [stack-name] # Create or update multiple stacks
-stack_master apply [region-or-alias] # Create or update stacks in the given region
+stack_master apply [environment] [stack-name] # Create or update a stack
+stack_master apply [environment] [stack-name] [environment] [stack-name] # Create or update multiple stacks
+stack_master apply [environment] # Create or update stacks in the given region
 stack_master apply # Create or update all stacks
 stack_master --changed apply # Create or update all stacks that have changed
-stack_master --yes apply [region-or-alias] [stack-name] # Create or update a stack non-interactively (forcing yes)
-stack_master diff [region-or-alias] [stack-name] # Display a stack tempalte and parameter diff
-stack_master delete [region-or-alias] [stack-name] # Delete a stack
-stack_master events [region-or-alias] [stack-name] # Display events for a stack
-stack_master outputs [region-or-alias] [stack-name] # Display outputs for a stack
-stack_master resources [region-or-alias] [stack-name] # Display outputs for a stack
+stack_master --yes apply [environment] [stack-name] # Create or update a stack non-interactively (forcing yes)
+stack_master diff [environment] [stack-name] # Display a stack tempalte and parameter diff
+stack_master delete [environment] [stack-name] # Delete a stack
+stack_master events [environment] [stack-name] # Display events for a stack
+stack_master outputs [environment] [stack-name] # Display outputs for a stack
+stack_master resources [environment] [stack-name] # Display outputs for a stack
```

Comment:
Are we changing the core API here? Can you continue to use StackMaster in the traditional way, i.e. by just specifying a region.

Reply:
there would be nothing stopping you giving environments the same names as the region they are in so yes.

However, yes this whole PR is a major breaking change. The yaml format is similar but different.

Mainly turns things around from stackmaster having regions as it's key data model entry point to environments.

Reply:
I feel we are making a change that breaks existing functionality for the benefit of some but not all users. This doesn't feel like the right trade off to me.

I like the idea of environments, I think moving away from region aliases is a good move but this feels like the wrong approach. What about a non breaking change like:

```yaml
stack_definitions:
  myapp-vpc:
    template: vpc.rb
  myapp-elb:
    template: elb.rb
  myapp-web:
    template: web.rb

environments:
  production:
    us-east-1:
      - myapp-vpc
      - myapp-elb
      - myapp-web
    eu-central-1:
      - myapp-vpc
      - myapp-elb
      - myapp-web
  staging:
    us-west-2:
      - myapp-vpc
      - myapp-elb
      - myapp-web
```

Reply:
This is better than what we have, that's for sure.  Right now I have no way to deploy the same stack  for different environments to the same region.